src/mpi/init/init_thread_cs.c

```diff
@@ -0,0 +1,185 @@
+/* -*- Mode: c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#include "mpidimpl.h"
+#include "mpi_init.h"
+
+#if defined(MPICH_IS_THREADED)
+
+#if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__GLOBAL || \
+    MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__POBJ   || \
+    MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__VCI
+MPID_Thread_mutex_t MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX;
+#endif
+
+#if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__POBJ
+MPID_Thread_mutex_t MPIR_THREAD_POBJ_HANDLE_MUTEX;
+MPID_Thread_mutex_t MPIR_THREAD_POBJ_MSGQ_MUTEX;
+MPID_Thread_mutex_t MPIR_THREAD_POBJ_COMPLETION_MUTEX;
+MPID_Thread_mutex_t MPIR_THREAD_POBJ_CTX_MUTEX;
+MPID_Thread_mutex_t MPIR_THREAD_POBJ_PMI_MUTEX;
+#endif
+
+#if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__VCI
+MPID_Thread_mutex_t MPIR_THREAD_POBJ_HANDLE_MUTEX;
+#endif
+
+static int cs_initialized = false;
+
+/* These routine handle any thread initialization that my be required */
+int MPID_thread_init(int *required)
```

Comment:
Perhaps these should have `MPII_` prefix? They are not device interfaces. Really, they are internal helpers only used during init.